app/uk/gov/hmrc/lisaapi/config/microserviceGlobal.scala

```diff
@@ -94,4 +97,17 @@ object MicroserviceGlobal extends DefaultMicroserviceGlobal with RunMode with Mi
 
   override def onHandlerNotFound(request: RequestHeader): Future[Result] = Future.successful(NotFound(Json.toJson(ErrorNotFound)))
 
+  override def onRouteRequest(request: RequestHeader): Option[Handler] = {
```

Comment:
We probably need to discuss this a little further - when sending a request to the service, I believe this would mean that any new APIs would produce a 404. This means the `validateHeader` check should never actually fail for an incorrect version. I don't know if there is a strong argument for this behaviour, or instead for the "incorrect version" error message behaviour?

Reply:
Yup lets discuss when you're in.

Other APIs on the platform work the same way as this but I can see an argument for being more explicit with our error messages.

We will need to approach the error message(s) carefully though - there's a difference between a version that's invalid (vONE), a version that doesn't exist (v99), or a version that is valid and exists, but doesn't contain the requested endpoint (i.e. withdrawal charge apis on v1.0).

Perhaps the existing - simpler - approach is the better approach initially, then we build on it based on feedback from real world usage?

Reply:
After discussion we've made went with the accept_header_invalid errors over the not found, and removed this routing feature.